### DISCUSSION TITLE
Firestore

### DISCUSSION THREAD

**Discussion by hansipie** [444B/streamlit-analytics2#59] [Q&A]
Hello
I can't get storage in firebase working ...
I have simplified my code to :
```python
    with streamlit_analytics.track(
        firestore_collection_name="analytics",
        firestore_key_file="firestore-key.json",
        ) as analytics:
        test = st.button("Test")
``` 
but i still get the following error: 
(...)
  File "C:\Users\hansi\Code\SmartComments\.venv\Lib\site-packages\google\cloud\firestore_v1\field_path.py", line 272, in __init__
    raise ValueError(error)
ValueError: One or more components is not a string or is empty.

The DB/collection 'analytics' is created. I've generated the json key file. The rule is in prod mode.
Any idea ? Am I missing something ?

**Comment by 444B** [comment]
@hansipie thanks for the question! My apologies for the late response, I didnt get notified of the discussion being raised. Will look into that

Ill look into replicating this tomorrow

In the meantime, line 4 strikes me as strange:
```python
        ) as analytics:
``` 
1. Can you expand on the need to provide an alias to the function and if this has worked successfully in other files?
2. Was this working in a previous version of streamlit-analytics? AFAIK, no code relating to the tracking functions has been changes in the fork but Ill double check that

**Comment by hansipie** [comment]
Hello,

This alias is not relevant... It's comming from some exemple i've found in the community. I've only forgot to remove it. However the issue remains without the alias.
